mentoring/light_children.py

```diff
@@ -101,12 +101,17 @@ def _is_default(value):
 
     @classmethod
     def init_block_from_node(cls, block, node, attr):
+        from mentoring import MentoringBlock
         block.light_children = []
         for child_id, xml_child in enumerate(node):
             cls.add_node_as_child(block, xml_child, child_id)
 
         for name, value in attr:
-            setattr(block, name, value)
+            if isinstance(block, MentoringBlock):
+                parsed_value = getattr(MentoringBlock, name).from_json(value)
```

Comment:
This would change the behavior - you're getting the attribute from the field instead of getting it from the function parameter. I would be very wary of making such a change without fully understanding the effects.

Also, if the goal is to ensure you do have a boolean, you don't need to change anything here - there is an XBlock field property we have contributed specifically for this type of issue, `enforce_type` - cf https://github.com/edx/XBlock/pull/200 . See https://github.com/edx-solutions/xblock-mentoring/blob/f4f37ca9a39dfdf6c58db0e3b1563738372bf13e/mentoring/mentoring.py#L99 for an example.